src/Integrations/Laravel/BackupManagerServiceProviderLaravel5.php

```diff
@@ -0,0 +1,166 @@
+<?php namespace BigName\BackupManager\Integrations\Laravel;
+
+use BigName\BackupManager\Databases;
+use BigName\BackupManager\Filesystems;
+use BigName\BackupManager\Compressors;
+use Symfony\Component\Process\Process;
+use Illuminate\Support\ServiceProvider;
+use BigName\BackupManager\Config\Config;
+use BigName\BackupManager\ShellProcessing\ShellProcessor;
+
+/**
+ * Class BackupManagerServiceProvider
+ * @package BigName\BackupManager\Integrations\Laravel
+ */
+class BackupManagerServiceProviderLaravel5 extends ServiceProvider
+{
+    protected $defer = true;
+
+    /**
+     * Bootstrap the application events.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        $configPath = __DIR__ . '/../../../config/';
+        $this->publishes([
+            $configPath => config_path('backup-manager/'),
+        ], 'backup-manager');
+    }
+
+    /**
+     * Register the service provider.
+     *
+     * @return void
+     */
+    public function register()
+    {
+
+        $this->mergeConfigFrom(config_path('backup-manager/storage.php'), 'backup-manager::storage');
+        $this->mergeConfigFrom(config_path('backup-manager/database.php'), 'backup-manager::database');
```

Comment:
@GrahamCampbell what do you recommend that the second param be?  I was trying not to muck with @ShawnMcCool code for the line:

```php
$provider = new Filesystems\FilesystemProvider(new Config($app['config']['backup-manager::storage']));
```

Again, can you please advise what you recommend here?  If you test this out you can see that it works during database backup and restore.  I tested only for AWS S3 bucket so far.